server/client.go

```diff
@@ -1428,13 +1454,28 @@ func (c *client) canSubscribe(subject []byte) bool {
 
 	// Check allow list. If no allow list that means all are allowed. Deny can overrule.
 	if c.perms.sub.allow != nil {
-		r := c.perms.sub.allow.Match(string(subject))
+		r := c.perms.sub.allow.Match(subject)
 		allowed = len(r.psubs) != 0
 	}
 	// If we have a deny list and we think we are allowed, check that as well.
 	if allowed && c.perms.sub.deny != nil {
-		r := c.perms.sub.deny.Match(string(subject))
+		r := c.perms.sub.deny.Match(subject)
 		allowed = len(r.psubs) == 0
+
+		// We use the actual subscription to signal us to spin up the deny mperms
+		// and cache. We check if the subject is a wildcard that contains any of
+		// the deny clauses.
+		// FIXME(dlc) - We could be smarter and track when these go away and remove.
+		if c.mperms == nil && subjectHasWildcard(subject) {
+			// Whip through the deny array and check if this wildcard subject is within scope.
+			for _, sub := range c.darray {
+				tokens := strings.Split(sub, tsep)
+				if isSubsetMatch(tokens, sub) {
+					c.loadMsgDenyFilter()
+					break
```

Comment:
The way I read this is that if there were deny clauses "foo.bar" and "foo.baz" and say subscription created is on ">", then we would add only "foo.bar" because of the `break`. I think we should be adding `foo.baz` too.

Reply:
When we trigger this action we load all deny subjects all at once and start checking all message deliveries.

Reply:
Ok, so if I understand correctly, when encountering the first subscription with a wildcard that match any of the deny clause, then we add all deny clauses to that mperms.deny sublist, right?

Reply:
Correct.